src/index.ts

```diff
@@ -119,6 +122,23 @@ module.exports.${groupName} = ${groupName}Definitions
   return format(lib, PRETTIER_OPTS);
 }
 
+function generateEsmLibrary(
+  definitionsByGroup: Map<string, Record<string, MessageDefinition>>,
+): string {
+  let lib = `${LICENSE}\n`;
+  for (const [groupName, definitions] of definitionsByGroup.entries()) {
+    lib += `
+const ${groupName}Definitions = ${JSON.stringify(definitions)}
+export { ${groupName}Definitions as ${groupName} }
+`;
+  }
+  lib += `export default { ${Array.from(definitionsByGroup.keys())
```

Comment:
I think this should be `export { ... }` to match the .d.ts?

Or more simply, `export const ${groupName} = ...`

Reply:
Nice catch, I did `export { ${groupName}Definitions as ${groupName} }` after each iteration already.

Howerver, In https://github.com/foxglove/studio/blob/main/packages/studio-base/src/players/FoxgloveWebSocketPlayer/index.ts#L13 it import with default which require export default as well. So there are two options.

1. I can edit `.d.ts` to export default
2. change studio-base import with no default, then I can remove export default. However, cjs module export may still have default export after convert.

I prefer 1st option

Let me know which we you prefer, I can update PR.

Reply:
I think 1 sounds fine as well. Do we need to change anything about the cjs module?

Reply:
done with `.d.ts` change, no need to change cjs module.